libr/anal/fcn.c

```diff
@@ -414,11 +415,35 @@ static bool is_delta_pointer_table (RAnal *anal, ut64 ptr) {
 	return true;
 }
 
+static bool regs_exist(RAnalValue *src, RAnalValue *dst) {
+	return src && dst && src->reg && dst->reg && src->reg->name && dst->reg->name;
+}
+
+//0 if not skipped; 1 if skipped; 2 if skipped before
+static int skip_hp(RAnal *anal, RAnalFunction *fcn, RAnalOp *op, RAnalBlock *bb, ut64 addr,
+                   char *tmp_buf, int oplen, int un_idx,int *idx) {
+	//this step is required in order to prevent infinite recursion in some cases
+	if ((addr + un_idx - oplen) == fcn->addr) {
+		if (!anal->flb.exist_at (anal->flb.f, "skip", 4, op->addr)) {
+			memset (tmp_buf + 5, 0, MAX_FLG_NAME_SIZE);
```

Comment:
It looks bad if I put this check in every if where I need it, so I prefer to keep it.